nbs/99_manuscript/coefs_comp/py/08_05-gtex_whole_blood-intersections_plots.py

```diff
@@ -0,0 +1,634 @@
+            coefficient.
+        query_set: a tuple with strings that specifies a query. For example
+            ("Clustermatch (high)", "Pearson (low") would select all gene pairs
+            for which clustermatch is high and pearson is low.
+
+    Returns:
+        A slice of variable "data_r_data" where the conditions specified in query_set
+        apply.
+    """
+    assert all([x in df_r_data_boolean_cols for x in query_set])
+
+    query = np.concatenate(
+        [
+            # columns that have to be true
+            np.concatenate(
+                [df_r_data[c].to_numpy().reshape(-1, 1) for c in query_set], axis=1
+            )
+            .all(axis=1)
+            .reshape(-1, 1),
+            # rest of the columns, that have to be false
+            np.concatenate(
+                [
+                    ~df_r_data[c].to_numpy().reshape(-1, 1)
+                    for c in df_r_data_boolean_cols
+                    if c not in query_set
+                ],
+                axis=1,
+            )
+            .all(axis=1)
+            .reshape(-1, 1),
+        ],
+        axis=1,
+    ).all(axis=1)
+
+    _tmp_df = df_r_data[query]
+
+    # sort by firt_coef value
+    _tmp_df = _tmp_df.sort_values(first_coef, ascending=False)
+
+    _tmp_df = _tmp_df[
+        [x for x in _tmp_df.columns if "(high)" not in x and "(low)" not in x]
+    ]
+
+    return _tmp_df
+
+
+# %%
+def plot_and_save_gene_pair(data, gene0_id, gene1_id, output_file_subset):
+    """
+    This function creates a joint plot of a pair of genes. It's used to to
+    select the final gene pairs to include in the paper.
+    """
+    gene0_symbol = gene_map[gene0_id]
+    gene1_symbol = gene_map[gene1_id]
+
+    with sns.plotting_context("paper", font_scale=2.0):
+        p = jointplot(
+            data,
+            x=gene0_id,
+            y=gene1_id,
+            add_corr_coefs=False,
+        )
+
+        p.ax_joint.set_xlabel(f"{gene0_symbol}", fontstyle="italic")
+        p.ax_joint.set_ylabel(f"{gene1_symbol}", fontstyle="italic")
+
+        output_file = (
+            OUTPUT_FIGURE_DIR
+            / f"genes-{output_file_subset}-{gene0_symbol}_vs_{gene1_symbol}.svg"
+        )
+        display(output_file)
+
+        plt.savefig(
+            output_file,
+            bbox_inches="tight",
+            dpi=300,
+            facecolor="white",
+        )
+
+
+# %% [markdown] tags=[]
+# ## Clustermatch/Spearman vs Pearson
+
+# %%
+_tmp_df = get_gene_pairs(
+    "clustermatch",
+    {
+        "Clustermatch (high)",
+        "Spearman (high)",
+        "Pearson (low)",
+    },
+)
+
+display(_tmp_df.shape)
+display(_tmp_df)
+
+# %% [markdown]
+# ### Preview
+
+# %%
+for i in range(min(_tmp_df.shape[0], 5)):
+    display(f"Index: {i}")
+    p = plot_gene_pair(_tmp_df, i)
+    display(p.fig)
+    plt.close(p.fig)
+
+# %% [markdown]
+# ### Selection
+
+# %%
+gene_pair_subset = "c_rs_vs_r"
```

Comment:
Great question, this part on the interpretability of results is sort of delicate. We haven't analyzed this particular pair too much though. I included it (selected) because it seems to be a non-coexistence relationship: looks like when *SDS* is highly expressed, *IFNG* is not, and vice-versa. In the figure I pasted above I show all the examples that we picked.